### DISCUSSION TITLE
Manual Focus in Cameraview

### DISCUSSION THREAD

**Discussion by fbeerens** [CommunityToolkit/Maui#2263] [New Feature Discussions]
Hi when using my camera on a Samsung Galaxy Tab A8, i can focus the camera by tapping the screen. I get a circle where to focus the camera. As soons as i use the camera view in my own made android application the this option is gone.
Can i activate that somehow or has anybode an idea how to do manual focus with the cameraview ?

![image](https://github.com/user-attachments/assets/74a9decd-7ee7-4edf-9990-62cc0073fe2e)

**Comment by Xopabyteh** [comment]
Hey, joining the convo to hopefully give it a lil more attention.

Our application would really benefit from the CameraView component having a way to programmatically set its focus.

A new `FocusMode` setting could be employed:
```cs
enum FocusMode
{
    Auto,
    Manual
}
```

Along with a new command on the CameraView itself
```cs
var cam = new CameraView();
cam.ManualFocus(relativeX: 0d, relativeY: 0d);
// relativeXY could be values from 0 - 1
// 0,0 would probably be the topleft corner and
// 1,1 would be the bottomright corner
```

If possible (I'm unsure whether the mobile platforms actually support this), it would be cool to also have info about the focus, like
```cs
cam.IsManuallyFocused <bool>
cam.OnManualFocusCompleted <event>
```

This obviously clashes with the native Focus API derived from VisualElement.cs, but the naming could be worked around...
